services/src/main/java/org/fao/geonet/api/groups/GroupApi.java

```diff
@@ -0,0 +1,170 @@
+public class GroupApi {
+    /** API logo note.*/
+    private static final String API_GET_LOGO_NOTE = "If last-modified header "
+        + "is present it is used to check if the logo has been modified since "
+        + "the header date. If it hasn't been modified returns an empty 304 Not"
+        + " Modified response. If modified returns the image. If the group has "
+        + "no logo then returns a transparent 1x1 px PNG image.";
+
+    /** Six hours in seconds. */
+    private static final int SIX_HOURS = 60 * 60 * 6;
+    /** Transparent 1x1 px PNG encoded in Base64. */
+    private static final String TRANSPARENT_1_X_1_PNG_BASE64 = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR"
+        + "42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==";
+    /** Transparent 1x1 px PNG. */
+    private static final byte[] TRANSPARENT_1_X_1_PNG = Base64.decodeBase64(TRANSPARENT_1_X_1_PNG_BASE64);
+
+
+    /**
+     * Writes the group logo image to the response. If no image is found it
+     * writes a 1x1 transparent PNG. If the request contain cache related
+     * headers it checks if the resource has changed and return a 304 Not
+     * Modified response if not changed.
+     * @param groupId the group identifier.
+     * @param webRequest the web request.
+     * @param response the servlet response.
+     * @throws ResourceNotFoundException if no group exists with groupId.
+     */
+    @ApiOperation(value = "Get the group logo image.",
+        nickname = "get",
+        notes = API_GET_LOGO_NOTE
+    )
+    @RequestMapping(value = "/{groupId}/logo", method = RequestMethod.GET)
+    public final void getGroupLogo(
+        @ApiParam(value = "Group identifier", required = true) @PathVariable(value = "groupId") final Integer groupId,
+        final WebRequest webRequest,
+        HttpServletResponse response) throws ResourceNotFoundException {
+
+        ServiceContext context = ServiceContext.get();
+        if (context == null) {
+            throw new RuntimeException("ServiceContext not available");
+        }
+
+        GroupRepository groupRepository = context.getBean(GroupRepository.class);
+
+
+        Group group = groupRepository.findOne(groupId);
+        if (group == null) {
+            throw new ResourceNotFoundException(String.format("Group with ID ' %d' not found in this catalog.", groupId));
+        }
+        try {
+            final Path logosDir = Resources.locateLogosDir(context);
+            final String logoUUID = group.getLogo();
+            Path imagePath = null;
+            FileTime lastModifiedTime = null;
+            if (StringUtils.isNotBlank(logoUUID) && !logoUUID.startsWith("http://") && !logoUUID.startsWith("https//")) {
+                imagePath = Resources.findImagePath(logoUUID,
+                    logosDir);
+                if (imagePath != null) {
+                    String ext;
+                    Path fileName = imagePath.getFileName();
+                    if (fileName != null) {
```

Comment:
To check also if Apache Commons FileNameUtils can be used for this filename/extension management.